src/chocolatey/infrastructure.app/commands/ChocolateyListCommand.cs

```diff
@@ -115,113 +98,66 @@ public virtual void configure_argument_parser(OptionSet optionSet, ChocolateyCon
-                        (ApplicationParameters.Features.UsePackageRepositoryOptimizations, configuration.Features.UsePackageRepositoryOptimizations.to_string(), deprecationNotice),
-                    option =>
-                    {
-                        if (option != null)
-                        {
-                            configuration.Features.UsePackageRepositoryOptimizations = false;
-                        }
-                    })
-                ;
+                     option => configuration.Verbose = option != null);
         }
 
-        public virtual void handle_additional_argument_parsing(IList<string> unparsedArguments, ChocolateyConfiguration configuration)
+        public virtual int count(ChocolateyConfiguration config)
         {
-            configuration.Input = string.Join(" ", unparsedArguments);
+            config.ListCommand.LocalOnly = true;
+            config.QuietOutput = true;
+
+            return _packageService.count_run(config);
         }
 
-        public virtual void handle_validation(ChocolateyConfiguration configuration)
+        public virtual void handle_additional_argument_parsing(IList<string> unparsedArguments, ChocolateyConfiguration configuration)
         {
-            if (!string.IsNullOrWhiteSpace(configuration.SourceCommand.Username) && string.IsNullOrWhiteSpace(configuration.SourceCommand.Password))
-            {
-                this.Log().Debug(ChocolateyLoggers.LogFileOnly, "Username '{0}' provided. Asking for password.".format_with(configuration.SourceCommand.Username));
-                System.Console.Write("User name '{0}' provided. Password: ".format_with(configuration.SourceCommand.Username));
-                configuration.SourceCommand.Password = InteractivePrompt.get_password(configuration.PromptForConfirmation);
-            }
+            var argumentsWithoutLocalOnly = new List<string>(unparsedArguments.Count);
 
-            if (configuration.ListCommand.PageSize < 1 || configuration.ListCommand.PageSize > 100)
+            foreach (var argument in unparsedArguments)
             {
-                var message = "The page size has been specified to be {0:N0} packages. The page size cannot be lower than 1 package, and no larger than 100 packages.".format_with(configuration.ListCommand.PageSize);
-                throw new ApplicationException(message);
+                if (_unsupportedArguments.Contains(argument.TrimStart('-'), StringComparer.OrdinalIgnoreCase))
+                {
+                    this.Log().Warn(ChocolateyLoggers.Important, @"
+UNSUPPORTED ARGUMENT: Ignoring the argument {0}. This argument is unsupported for locally installed packages, and will be treated as a package name in Chocolatey CLI v3!", argument);
+                }
+                else
+                {
+                    argumentsWithoutLocalOnly.Add(argument);
+                }
             }
+
+            configuration.Input = string.Join(" ", argumentsWithoutLocalOnly);

```

Comment:
Is a package name with a _prefixed_ `-` character valid? The `install`, `uninstall`, and `upgrade` commands _all_ filter out any arguments that have a `-` prefix before running their operations on the remaining unparsed arguments.

For consistency, I think we should do the same here.

Reply:
Actually, ignore this; this is the same as the current list/search behaviour, no point changing it right now.